app/src/userpages/modules/permission/reducer.js

```diff
@@ -16,6 +16,18 @@ const initialState = {
     fetching: false,
 }
 
+const newPermissions = (state, type, id, nextPermissions) => {
```

Comment:
I'd rename this maybe like `getNewPermissions`

Reply:
Signature could be just `const getNewPermissions = (prevPermissions, nextPermissions)`

```js
getNewPermissions(
    (state.byTypeAndId[action.resourceType] || {})[action.resourceId],
    action.permissions.map(({ operation }) => operation).sort(),
)
```